gobblin-core/src/main/java/org/apache/gobblin/publisher/BaseDataPublisher.java

```diff
@@ -487,6 +480,15 @@ protected void addSingleTaskWriterOutputToExistingDir(Path writerOutputDir, Path
     }
   }
 
+  protected void addWriterOutputToNewDir(Path writerOutputDir, Path publisherOutputDir,
+      WorkUnitState workUnitState, int branchId, ParallelRunner parallelRunner)
+      throws IOException {
+    // Create the parent directory of the final output directory if it does not exist
+    WriterUtils.mkdirsWithRecursivePermissionWithRetry(this.publisherFileSystemByBranches.get(branchId),
+            publisherOutputDir.getParent(), this.permissions.get(branchId), retrierConfig);
```

Comment:
Is there a reason why we're omitting the set output dir groups? I believe it's needed for permissions if configured
```
if(this.publisherOutputDirOwnerGroupByBranches.get(branchId).isPresent()) {
          LOG.info(String.format("Setting path %s group to %s", publisherOutputDir.toString(), this.publisherOutputDirOwnerGroupByBranches.get(branchId).get()));
          HadoopUtils.setGroup(this.publisherFileSystemByBranches.get(branchId), publisherOutputDir, this.publisherOutputDirOwnerGroupByBranches.get(branchId).get());
 ```

Reply:
wow thank you for the catch, I didn't notice and should have been more careful. I just added the missing lines to the new functions.